ENGAGEHF/Education/Views/VideoPlayers/VideoPlayer.swift

```diff
@@ -0,0 +1,36 @@
+//
+// This source file is part of the ENGAGE-HF project based on the Stanford Spezi Template Application project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+import SpeziViews
+import SwiftUI
+
+
+struct VideoPlayer: View {
+    let youtubeId: String
+    
+    @State private var viewState: ViewState = .processing
+    
+    
+    var body: some View {
+        ZStack {
+            WebView(urlString: "https://youtube.com/embed/\(youtubeId)", viewState: $viewState)
+                .aspectRatio(16 / 9, contentMode: .fit)
+            if viewState == .processing {
+                ProgressView()
+                    .background(.clear)
+            }
```

Comment:
I would suggest that we use the video thumbnail view here as a placeholder behind the simple progress view.

We already have the image loaded in the other view so ideally it is cached locally or even loads way faster than the actual web view (as we see in the list where they load quite fast).

I can imagine it looks nice if it uses a 0.5 occupancy to translate that it is loading. Or maybe a pulsing effect like something described here between 0.2 and 0.5?
https://stackoverflow.com/a/78412804